pkg/policy/mapstate_test.go

```diff
@@ -727,57 +739,338 @@ func (ds *PolicyTestSuite) TestMapState_DenyPreferredInsert(c *check.C) {
 	}
 }
 
-func (ds *PolicyTestSuite) TestMapState_AccumulateMapChanges(c *check.C) {
-	csFoo := newTestCachedSelector("Foo", false)
-	csBar := newTestCachedSelector("Bar", false)
-
-	TestKey := func(id int, port uint16, proto uint8, direction trafficdirection.TrafficDirection) Key {
-		return Key{
-			Identity:         uint32(id),
-			DestPort:         port,
-			Nexthdr:          proto,
-			TrafficDirection: direction.Uint8(),
-		}
-	}
-	TestIngressKey := func(id int, port uint16, proto uint8) Key {
-		return TestKey(id, port, proto, trafficdirection.Ingress)
-	}
-	TestEgressKey := func(id int, port uint16, proto uint8) Key {
-		return TestKey(id, port, proto, trafficdirection.Egress)
-	}
-	DNSUDPEgressKey := func(id int) Key {
-		return TestEgressKey(id, 53, 17)
+func testKey(id int, port uint16, proto uint8, direction trafficdirection.TrafficDirection) Key {
+	return Key{
+		Identity:         uint32(id),
+		DestPort:         port,
+		Nexthdr:          proto,
+		TrafficDirection: direction.Uint8(),
 	}
-	DNSTCPEgressKey := func(id int) Key {
-		return TestEgressKey(id, 53, 6)
+}
+
+func testIngressKey(id int, port uint16, proto uint8) Key {
+	return testKey(id, port, proto, trafficdirection.Ingress)
+}
+
+func testEgressKey(id int, port uint16, proto uint8) Key {
+	return testKey(id, port, proto, trafficdirection.Egress)
+}
+
+func DNSUDPEgressKey(id int) Key {
+	return testEgressKey(id, 53, 17)
+}
+
+func DNSTCPEgressKey(id int) Key {
+	return testEgressKey(id, 53, 6)
+}
+
+func HostIngressKey() Key {
+	return testIngressKey(1, 0, 0)
+}
+
+func AnyIngressKey() Key {
+	return testIngressKey(0, 0, 0)
+}
+
+func AnyEgressKey() Key {
+	return testEgressKey(0, 0, 0)
+}
+
+func HttpIngressKey(id int) Key {
+	return testIngressKey(id, 80, 6)
+}
+
+func HttpEgressKey(id int) Key {
+	return testEgressKey(id, 80, 6)
+}
+
+func testEntry(proxyPort uint16, deny bool, owners ...MapStateOwner) MapStateEntry {
+	entry := MapStateEntry{
+		ProxyPort: proxyPort,
+		IsDeny:    deny,
 	}
-	HostIngressKey := func() Key {
-		return TestIngressKey(1, 0, 0)
+	if len(owners) > 0 {
+		entry.owners = make(map[MapStateOwner]struct{}, len(owners))
 	}
-	AnyIngressKey := func() Key {
-		return TestIngressKey(0, 0, 0)
+	for _, owner := range owners {
+		entry.owners[owner] = struct{}{}
 	}
-	//AnyEgressKey := func() Key {
-	//	return TestEgressKey(0, 0, 0)
-	//}
-	HttpIngressKey := func(id int) Key {
-		return TestIngressKey(id, 80, 6)
+	return entry
+}
```

Comment:
For what it's worth, I found this easier to read by adding these small changes:

```
diff --git a/pkg/policy/mapstate_test.go b/pkg/policy/mapstate_test.go
index ab6b0e3cb41d..b382e847d3bd 100644
--- a/pkg/policy/mapstate_test.go
+++ b/pkg/policy/mapstate_test.go
@@ -798,6 +798,14 @@ func testEntry(proxyPort uint16, deny bool, owners ...MapStateOwner) MapStateEnt
        return entry
 }
 
+func allowEntry(proxyPort uint16, owners ...MapStateOwner) MapStateEntry {
+       return testEntry(proxyport, false, owners...)
+}
+
+func denyEntry(proxyPort uint16, owners ...MapStateOwner) MapStateEntry {
+       return testEntry(proxyport, true, owners...)
+}
+
 func testEntryD(proxyPort uint16, deny bool, derivedFrom labels.LabelArrayList, owners ...MapStateOwner) MapStateEntry {
        entry := testEntry(proxyPort, deny, owners...)
        entry.DerivedFromRules = derivedFrom
```

....and then doing a couple of sed expressions on the rest of the file so that I could read each test as a series of `allowEntry` / `denyEntry` callouts.

If you think this makes it easier, I'd be in favour of adding this change in, but not required :)

Reply:
Good point, will adopt :-)

Reply:
Done.